lib/takeaway.rb

```diff
@@ -0,0 +1,33 @@
+require_relative 'takeaway_menu'
+require_relative 'basket'
+require_relative 'messager'
+
+class TakeAway
+
+  def initialize(
+    menu_class = TakeAwayMenu, menu = menu_class.new, basket = Basket.new(menu)
+  )
+    @menu = menu
+    @basket = basket
+  end
+
+  def read_menu
+    @menu.read
+  end
+
+  def add_to_basket(dish)
+    @basket.add(dish)
+    "#{dish} added to basket"
+  end
+
+  def view_basket
+    @basket.view
+  end
+
+  def complete_order(customer_number, messager = Messager.new(customer_number),
+    time_in_30 = (Time.now + 30 * 60).strftime("%I:%M%p"))
```

Comment:
better to indent this line so it aligns with the first parameter on the line above to make it clearer that you've split the parameters over two lines